### PR TITLE
Doc v2 c1

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/math/regular/cos.hpp [hunk 1]
```diff
@@ -48,36 +48,47 @@ namespace eve
...
-//!
-//!     provide a balance between speed and range limitation.
-//!
-//!  @}
+//!        * If the element is \f$\pm0\f$, \f$1\f$ is returned.
+//!        * If the element is \f$\pm\infty\f$, Nan is returned.
+//!        * If the element is a `Nan`, `Nan` is returned.
+//!    2. [The operation is performed conditionnaly](@ref conditional).
+//!    3. These are optimized calls providing a balance between speed and range limitation.
+//!        1. assumes that the inputs elements  belong to \f$[-\pi/4,\pi/4]\f$ and return NaN outside.
```
Comment (jfalcou) [inline include/eve/module/math/regular/cos.hpp:80]: shouldn't it be a. b. c. ?

File: include/eve/module/math/regular/coth.hpp [hunk 2]
```diff
@@ -42,30 +41,32 @@ namespace eve
...
+//!     * `c`: [Conditional expression](@ref conditional_expr) masking the operation.
+//!     * `m`: [Logical value](@ref logical) masking the operation.
 //!
-//!      In particular:
+//! **Return value**
 //!
-//!      * If the element is \f$\pm0\f$, \f$\pm\infty\f$ is returned.
-//!      * If the element is \f$\pm\infty\f$, \f$\pm1\f$ is returned.
-//!      * If the element is a `NaN`, `NaN` is returned.
+//!   1.  Returns the [elementwise](@ref glossary_elementwise) hyperbolic cosine of the input.
```
Comment (jfalcou) [inline include/eve/module/math/regular/coth.hpp:61]: cotangent ?

File: include/eve/module/math/regular/csch.hpp [hunk 3]
```diff
@@ -42,26 +41,32 @@ namespace eve
...
+//!      * `x`: [floating value](@ref floating_value).
+//!      * `c`: [Conditional expression](@ref conditional_expr) masking the operation.
+//!      * `m`: [Logical value](@ref logical) masking the operation.
 //!
-//! **Return value**
+//!   **Return value**
 //!
-//!   *  Returns the [elementwise](@ref glossary_elementwise) hyperbolic secant of the input.
-//!      In particular:
+//!    1.  Returns the [elementwise](@ref glossary_elementwise) hyperbolic secant of the input.
```
Comment (jfalcou) [inline include/eve/module/math/regular/csch.hpp:61]: cosecant ?

Comment (jtlap) [reply include/eve/module/math/regular/csch.hpp:61]: yes

File: include/eve/module/math/regular/coth.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/math/regular/coth.hpp:61]: yes

**Pull request merged**